File: mock_HUB_Sender.py
#Use this to test the connection, the data it will accept is GPS long and lat.
import requests

def send_gps_data(longitude, latitude, api_gateway_url):
    data = {
        'longitude': longitude,
        'latitude': latitude
    }
    headers = {'Content-Type': 'application/json'}

    try:
        response = requests.post(api_gateway_url, json=data, headers=headers)
        response.raise_for_status()  # Raise an exception for bad status codes

        print("GPS data sent successfully!")
        if response.content:  # Check if there's a response body
            print("API Gateway response:", response.json()) 

    except requests.exceptions.RequestException as e:
        print(f"Failed to send GPS data. Error: {e}")

# Example usage
api_gateway_url = 'https://fnh2s8vbnl.execute-api.ap-southeast-2.amazonaws.com/receiver'  # Replace with your actual API Gateway URL
send_gps_data(-122.3321, 47.6062, api_gateway_url)

File: mock_HUB_sender.cpp
#include <ESP8266WiFi.h>
#include <ESP8266HTTPClient.h>
#include <ArduinoJson.h>

const char* ssid = "your_wifi_ssid";
const char* password = "your_wifi_password";
const char*   
 api_gateway_url = "https://your-api-id.execute-api.your-region.amazonaws.com/your-stage/gps";

void setup() {
  Serial.begin(115200);
  WiFi.begin(ssid, password);

  while (WiFi.status() != WL_CONNECTED) {
    delay(1000);
    Serial.println("Connecting to WiFi...");
  }
  Serial.println("Connected to WiFi");   

}

void loop() {
  // Get GPS data (replace with your actual GPS reading logic)
  float latitude = 47.6062; 
  float longitude = -122.3321;

  if (WiFi.status() == WL_CONNECTED) {
    HTTPClient http;
    http.begin(api_gateway_url);
    http.addHeader("Content-Type", "application/json");

    StaticJsonDocument doc;
    doc["latitude"] = latitude;
    doc["longitude"] = longitude;
    String json_data;
    serializeJson(doc, json_data);

    int httpResponseCode = http.POST(json_data);

    if (httpResponseCode > 0) {
      Serial.print("HTTP Response code: ");
      Serial.println(httpResponseCode);
      String payload = http.getString();
      Serial.println(payload);
    } else {
      Serial.print("Error code: ");
      Serial.println(httpResponseCode);
    }
    http.end();
  }

  delay(5000);   
 // Send data every 5 seconds (adjust as needed)
}
